pkg/liqo-controller-manager/namespaceMap-controller/manage_remoteNamespaces.go

```diff
@@ -58,49 +59,58 @@ func (r *NamespaceMapReconciler) deleteRemoteNamespace(clusterID, remoteName str
 }
 
 // This function checks if there are Namespaces that have to be created or deleted, in accordance with DesiredMapping
-// and CurrentMapping fields.
-func (r *NamespaceMapReconciler) manageRemoteNamespaces(nm *mapsv1alpha1.NamespaceMap) error {
+// and CurrentMapping fields. It updates also NamespaceOffloading status in according to the remote Namespace Status.
+func (r *NamespaceMapReconciler) manageRemoteNamespaces(nm *mapsv1alpha1.NamespaceMap, requeue *bool) error {
 	if nm.Status.CurrentMapping == nil {
 		nm.Status.CurrentMapping = map[string]mapsv1alpha1.RemoteNamespaceStatus{}
 	}
 
 	// if DesiredMapping field has more entries than CurrentMapping, is necessary to create new remote namespaces
 	for localName, remoteName := range nm.Spec.DesiredMapping {
-		if _, ok := nm.Status.CurrentMapping[localName]; ok {
+		if remoteStatus, ok := nm.Status.CurrentMapping[localName]; ok && remoteStatus.Phase == mapsv1alpha1.MappingAccepted {
 			continue
 		}
+		namespacePhase := mapsv1alpha1.MappingAccepted
+		message := "Your remote namespace is correctly created on this cluster"
 		if err := r.createRemoteNamespace(nm.Labels[liqoconst.RemoteClusterID], remoteName); err != nil {
-			return err
+			namespacePhase = mapsv1alpha1.MappingCreationLoopBackOff
+			message = fmt.Sprintf("%s", err)
+			*requeue = true
+			klog.Errorf("%s -> Namespace '%s' cannot be created on cluster '%s'",
+				err, localName, nm.Labels[liqoconst.RemoteClusterID])
 		}
 		nm.Status.CurrentMapping[localName] = mapsv1alpha1.RemoteNamespaceStatus{
 			RemoteNamespace: remoteName,
-			Phase:           mapsv1alpha1.MappingAccepted,
+			Phase:           namespacePhase,
 		}
-		if err := r.Update(context.TODO(), nm); err != nil {
-			klog.Errorf("unable to update NamespaceMap '%s' Status", nm.Name)
+
+		if err := r.setReadyConditionOnRemoteNamespace(localName, message,
+			nm.Labels[liqoconst.RemoteClusterID], namespacePhase); err != nil {
 			return err
 		}
-		klog.Infof("Status of NamespaceMap '%s' is correctly updated, new remote Namespace '%s'",
-			nm.Name, remoteName)
 	}
 
-	// if DesiredMapping field has less entries than CurrentMapping, is necessary to remove some remote namespaces
+	// If DesiredMapping field has less entries than CurrentMapping, is necessary to remove some remote namespaces
 	for localName, remoteStatus := range nm.Status.CurrentMapping {
 		if _, ok := nm.Spec.DesiredMapping[localName]; ok {
 			continue
 		}
-		if err := r.deleteRemoteNamespace(nm.Labels[liqoconst.RemoteClusterID],
-			remoteStatus.RemoteNamespace); err != nil {
-			return err
+		// If exist, delete remote Namespaces
+		if remoteStatus.Phase == mapsv1alpha1.MappingAccepted {
+			if err := r.deleteRemoteNamespace(nm.Labels[liqoconst.RemoteClusterID],
+				remoteStatus.RemoteNamespace); err != nil {
+				return err
+			}
+			klog.Infof("Remote Namespace '%s' successfully deleted.", remoteStatus.RemoteNamespace)
```

Comment:
can we add more detailed logging here? For example, adding the local namespace name and the remote cluster id?

Reply:
yes sure

Reply:
sorry the right log is at line 57, this is a duplicate i have removed it, thank you